players/group_3.py

```diff
@@ -49,15 +57,33 @@ def init_cuts(self, current_percept) -> Tuple[List[List[int]]]:
         ([[y1, y1], [y2, y2]], [[x1, x1], [x2, x2]])          # i.e., (HORIZONTAL_CUTS, VERTICAL_CUTS)
         """
         cake_len, cake_width = current_percept.cake_len, current_percept.cake_width
-        row_count, col_count = 2, 4
-        row_height, col_width = cake_len / row_count, cake_width / col_count
-
-        horizontal_cuts = [[i * row_height, i * row_height] for i in range(1, row_count)]
-        vertical_cuts = [[i * col_width, i * col_width] for i in range(1, col_count)]
 
-        print(f"{cake_len=}, {cake_width=}")
-        print(f"init_{horizontal_cuts=}, {vertical_cuts=}")
-        return horizontal_cuts, vertical_cuts
+        working_pairs = []
+        extra_pieces = 0
+        extra_area = cake_len * cake_width * 0.05
+        while not working_pairs:
```

Comment:
@eddiechoi00 
if i understand correctly..
- first iteration in while loop : not using any extra pieces
- 2nd iteration : using 1 extra piece with area = extra area
- 3rd iteration : using 2 extra pieces each with area = extra area/2
- ...

break out of while-loop if at least one set of row/col pair that works is found

Q: what's the worst case scenario? can it ever be stuck in an infinite loop? (probably not given the constraint on request size)

---

(brainfart) justification for prioritizing less 'extra piece usage':
- 0 extra pieces is better than 1+ : better to use the extra space to enable for flexibility for each real cake slice cut rather than throwing away a whole piece just to fit the row/col structure
- `k` extra pieces is better than `k+1` extra pieces: with each additional fake cake slice, more real cake slices are affected. e.g., 1 fake cake slice can in the downsteram process impact the cut that affects all cake slices in the same row and/or column. 2 fake slices might affect cakes that are on (upto) 2 rows/cols. Given that each cut is calculated based on 'compromise' of the cakes affected by the cut and the descent (i.e., cut determination) doesn't distinguish between real and fake cake, having less fake cakes added in the structure is more beneficial in terms of reducing negative influence on other real cakes. (i.e., the number of real cakes affected is proportional to the number of fake cakes added) 

i just jotted my thoughts to see if i'm aligned and bc it might be something to include in the report.

Reply:
yea sounds good
not sure if it'll ever get stuck in an infinite loop, the greater the length of the requests the more likely itll have enough viable factors
id assume we'll mostly end up with 0 or 1 extra pieces but would have to run a sim to make sure